src/main/java/build/buildfarm/instance/shard/ShardInstance.java

```diff
@@ -478,6 +531,9 @@ public void stop() throws InterruptedException {
     if (dispatchedMonitor != null) {
       dispatchedMonitor.stop();
     }
+    if (prometheusMetricsThread != null) {
+      prometheusMetricsThread.stop();
```

Comment:
`Thread.stop()` is deprecated and `interrupt()` might be the method we want to use.

Reply:
https://docs.oracle.com/javase/8/docs/api/java/lang/Thread.html#stop--